lib/change_stream_simulator.js

```diff
@@ -0,0 +1,244 @@
+var Timestamp = require('mongodb-core').BSON.Timestamp;
+var MongoNetworkError = require('mongodb-core').MongoNetworkError;
+var supportedStages =  ['$match', '$project', '$addFields', '$replaceRoot', '$redact'];
+
+var createChangeStreamCursorSimulation = function (serverConfig, namespace, pipeline, options) {
```

Comment:
its not exactly super important for the future, but for the sake of review it would be nice to document these.